front/src/applications/operationalStudies/components/MacroEditor/ngeToOsrd.ts

```diff
@@ -258,155 +263,376 @@ const createTrainSchedulePayload = async ({
+): Promise<TimetableItemWithTimetableId> => {
+  if (isPacedTrainResponseWithPacedTrainId(timetableItem)) {
+    // If the TimetableItem is initially a PacedTrain
+    if (trainrunFrequency.id === 1) {
+      // If the frequency indicates that the TimetableItem is now a TrainSchedule
+      await dispatch(
+        osrdEditoastApi.endpoints.deletePacedTrain.initiate({
+          body: { ids: [formatPacedTrainIdToEditoastTrainId(timetableItem.id)] },
+        })
+      );
+      await dispatch(
+        osrdEditoastApi.endpoints.postTimetableByIdTrainSchedules.initiate({
+          id: timetableId,
+          body: [timetableItem],
+        })
+      ).unwrap();
+    } else {
+      // If the frequency indicates that the TimetableItem is still a PacedTrain
+      await dispatch(
+        osrdEditoastApi.endpoints.putPacedTrainById.initiate({
+          id: formatPacedTrainIdToEditoastTrainId(timetableItem.id),
+          body: {
+            ...timetableItem,
+            // Reset margins because they contain references to path items
+            margins: undefined,
+          },
+        })
+      ).unwrap();
+    }
+    const newPacedTrain = await fetchTimetableItem(timetableItem.id, dispatch);
+    return newPacedTrain;
+  }
+  // If the TimetableItem is initially a TrainSchedule
+  if (trainrunFrequency.id === 1) {
+    // If the frequency indicates that the TimetableItem is still a TrainSchedule
+    const newTrainSchedule = await dispatch(
+      osrdEditoastApi.endpoints.putTrainScheduleById.initiate({
+        id: formatTrainScheduleIdToEditoastTrainId(timetableItem.id),
+        trainScheduleForm: {
+          ...timetableItem,
+          // Reset margins because they contain references to path items
+          margins: undefined,
+        },
+      })
+    ).unwrap();
+    return { ...newTrainSchedule, id: timetableItem.id };
+  }
+  // If the frequency indicates that the TimetableItem is now a PacedTrain
+  await dispatch(
+    osrdEditoastApi.endpoints.deleteTrainSchedule.initiate({
+      body: { ids: [formatTrainScheduleIdToEditoastTrainId(timetableItem.id)] },
+    })
+  );
+  await dispatch(
+    osrdEditoastApi.endpoints.postTimetableByIdPacedTrains.initiate({
+      id: timetableId,
+      body: [timetableItem as PacedTrain],
+    })
+  ).unwrap();
+  const newPacedTrain = await fetchTimetableItem(timetableItem.id, dispatch);
+  return newPacedTrain;
+};
+
+const handleCreateTimetableItem = async (
+  timetableId: number,
+  netzgrafikDto: NetzgrafikDto,
+  trainrun: TrainrunDto,
+  infraId: number,
+  dispatch: AppDispatch,
+  addUpsertedTimetableItems: (timetableItems: TimetableItemWithTimetableId[]) => void,
+  state: MacroEditorState
+): Promise<void> => {
+  const { path, trainrunLabels, startDate, schedule } = await generateTrainrunData(
+    getTrainrunSectionsByTrainrunId(netzgrafikDto, trainrun.id),
+    netzgrafikDto.nodes,
+    trainrun,
+    infraId,
+    dispatch,
+    netzgrafikDto.labels
+  );
```

Comment:
You pass a lot of props from `netzgrafikDto` in this function, why not simply give the whole `netzgrafikDto` ? (and move the call to `getTrainrunSectionsByTrainrunId` inside `generateTrainrunData` ?)
```suggestion
  const { path, trainrunLabels, startDate, schedule } = await generateTrainrunData(
    netzgrafikDto
    trainrun,
    infraId,
    dispatch,
  );
```

Reply:
Done in "pass netzgrafikDto to generateTrainrunProperties"